src/test/java/org/broadinstitute/hellbender/tools/copynumber/PostprocessGermlineCNVCallsIntegrationTest.java

```diff
@@ -1,73 +1,209 @@
+        }
+
+        runCommandLine(argumentsBuilder.getArgsList());
+    }
+
+    @Test(dataProvider = "differentValidInput", groups = {"python"})
+    public void testDifferentValidInputIntervalsComplete(final List<String> callShards,
+                                                         final List<String> modelShards) throws IOException {
+        for (int sampleIndex = 0; sampleIndex < NUM_TEST_SAMPLES; sampleIndex++) {
+            final File actualIntervalsOutputVCF = createTempFile("intervals-output-vcf-" + sampleIndex, ".vcf");
+            final File actualSegmentsOutputVCF = createTempFile("segments-output-vcf-" + sampleIndex, ".vcf");
+            final File expectedIntervalsOutputVCF = INTERVALS_VCF_CORRECT_OUTPUTS.get(sampleIndex);
+            final File expectedSegmentsOutputVCF = SEGMENTS_VCF_CORRECT_OUTPUTS.get(sampleIndex);
+            runToolForSingleSample(callShards, modelShards, sampleIndex,
+                    actualIntervalsOutputVCF, actualSegmentsOutputVCF,
+                    ALLOSOMAL_CONTIGS, AUTOSOMAL_REF_COPY_NUMBER, false);
+            IntegrationTestSpec.assertEqualTextFiles(actualIntervalsOutputVCF, expectedIntervalsOutputVCF);
+            IntegrationTestSpec.assertEqualTextFiles(actualSegmentsOutputVCF, expectedSegmentsOutputVCF);
+        }
+    }
+
+    @Test(dataProvider = "differentValidInput")
+    public void testDifferentValidInputIntervalsVCFOnly(final List<String> callShards,
+                                                        final List<String> modelShards) throws IOException {
+        for (int sampleIndex = 0; sampleIndex < NUM_TEST_SAMPLES; sampleIndex++) {
+            final File actualIntervalsOutputVCF = createTempFile("intervals-output-vcf-" + sampleIndex, ".vcf");
+            final File expectedIntervalsOutputVCF = INTERVALS_VCF_CORRECT_OUTPUTS.get(sampleIndex);
+            runToolForSingleSample(callShards, modelShards, sampleIndex,
+                    actualIntervalsOutputVCF, null,
+                    ALLOSOMAL_CONTIGS, AUTOSOMAL_REF_COPY_NUMBER, false);
+            IntegrationTestSpec.assertEqualTextFiles(actualIntervalsOutputVCF, expectedIntervalsOutputVCF);
+        }
+    }
+
+    @Test(dataProvider = "differentInvalidInput")
+    public void testDifferentInvalidInput(final List<String> callShards,
+                                          final List<String> modelShards) throws IOException {
+        /* dry run -- just input data validation */
+        for (int sampleIndex = 0; sampleIndex < NUM_TEST_SAMPLES; sampleIndex++) {
+            try {
+                runToolForSingleSample(callShards, modelShards, sampleIndex,
+                        createTempFile("intervals-output-vcf", ".vcf"),
+                        createTempFile("segments-output-vcf", ".vcf"),
+                        ALLOSOMAL_CONTIGS, AUTOSOMAL_REF_COPY_NUMBER, true);
+            } catch (final IllegalArgumentException ex) {
```

Comment:
Why not use `expectedExceptions`?

Reply:
Because there is an internal loop and they all must throw. If we use `expectedExceptions`, the first exception is considered a success. The other option is unravel the sample index loop as different test cases, which now I did.

Reply:
FYI, I just discovered Assert.assertThrows, which might be useful in such cases in the future.